fairing/notebook/jobs.py

```diff
@@ -0,0 +1,61 @@
+import fairing
+from fairing.functions.function_shim import get_execution_obj_type, ObjectType
+from fairing.preprocessors.function import FunctionPreProcessor
+from fairing.builders.docker.docker import DockerBuilder
+from fairing.deployers.job.job import Job
+from fairing.deployers.serving.serving import Serving
+
+def guess_preprocessor(obj, *args, **kwargs):
+    if get_execution_obj_type(obj) != ObjectType.NOT_SUPPORTED:
+        return FunctionPreProcessor(obj, *args, **kwargs) 
+    else:
+        raise NotImplementedError("obj param should be a function or a class, got {}".format(type(obj)))
+
+def guess_docker_registry():
+    try:
+        gcp_project = fairing.cloud.gcp.guess_project_name()
+        registry = 'gcr.io/{}'.format(gcp_project)
+        return registry
+    except Exception as e:
+        return None
+
+class BaseJob:
```

Comment:
these classes should live somewhere outside the notebook package, possibly their own

Reply:
Ack.